File: src/main/java/com/legacy/app/InventoryManager.java
package com.legacy.app;

import java.util.ArrayList;
import java.util.List;
import org.json.simple.JSONObject;

public class InventoryManager {
    public List<String> p = new ArrayList<>(); // products
    public List<Double> pr = new ArrayList<>(); // prices
    public List<Integer> q = new ArrayList<>(); // quantities

    public void addForTests() {
        p.add("Widget"); pr.add(10.0); q.add(5);
        p.add("Gadget"); pr.add(20.0); q.add(3);
        p.add("Widget"); pr.add(10.0); q.add(2); // Duplicate
    }

    // Spaghetti code: confusing variable names, nested loops
    public void processProducts() {
        System.out.println("Processing products...");
        
        for (int i = 0; i < p.size(); i++) {
            String a = p.get(i);
            Double b = pr.get(i);
            Integer c = q.get(i);
            
            boolean d = false;
            // Inefficient O(n^2) check
            for (int k = 0; k < p.size(); k++) {
                if (i != k && p.get(k).equals(a)) {
                     d = true;
                }
            }

            if (d) {
                System.out.println("Duplicate found: " + a);
            }
            
            JSONObject obj = new JSONObject();
            obj.put("name", a);
            obj.put("val", b * c);
            System.out.println(obj);
        }
    }

    public double getTotalStockValue() {
        double total = 0.0;
        double tax = 0.05; // 5% tax or something

        for (int i = 0; i < p.size(); i++) {
            total += pr.get(i) * q.get(i);
        }

        // Logic Error: Subtracting tax instead of adding it (or just calculating total + tax)
        // Assuming we wanted Total + Tax
        return total - (total * tax); 
    }
}

File: nodejs/src/inventoryManager.js
class InventoryManager {
    constructor() {
        this.p = []; // products
        this.pr = []; // prices
        this.q = []; // quantities
    }

    addForTests() {
        this.p.push("Widget"); this.pr.push(10.0); this.q.push(5);
        this.p.push("Gadget"); this.pr.push(20.0); this.q.push(3);
        this.p.push("Widget"); this.pr.push(10.0); this.q.push(2); // Duplicate
    }

    // Spaghetti code: confusing variable names, nested loops
    processProducts() {
        console.log("Processing products...");

        for (let i = 0; i < this.p.length; i++) {
            let a = this.p[i];
            let b = this.pr[i];
            let c = this.q[i];

            let d = false;
            // Inefficient O(n^2) check
            for (let k = 0; k < this.p.length; k++) {
                if (i !== k && this.p[k] === a) {
                    d = true;
                }
            }

            if (d) {
                console.log("Duplicate found: " + a);
            }

            let obj = {};
            obj["name"] = a;
            obj["val"] = b * c; // Note: In Java code it was b * c.
            console.log(JSON.stringify(obj)); // mimicking System.out.println(obj) which usually prints JSON string if using json-simple
        }
    }

    getTotalStockValue() {
        let total = 0.0;
        let tax = 0.05; // 5% tax or something

        for (let i = 0; i < this.p.length; i++) {
            total += this.pr[i] * this.q[i];
        }

        // Logic Error: Subtracting tax instead of adding it (or just calculating total + tax)
        // Assuming we wanted Total + Tax
        return total - (total * tax);
    }
}

module.exports = InventoryManager;
